mbed-os/drivers/CAN.cpp

```diff
@@ -82,6 +82,11 @@ int CAN::read(CANMessage &msg, int handle)
     return ret;
 }
 
+int CAN::readNonLocking(CANMessage &msg, int handle)
```

Comment:
see my comment for this review, looks like this can be done without needing to do that can read from the irq and without needing to delete the mutex

Reply:
I don't think that's a good idea. We will likely run into the same issue.

Reply:
Didnungo thru the pr comment?

Reply:
What issue? With the implementation I linked, it should perform exactly the same as this

Reply:
I read the commend, its not the same... if you set a flag on interrupt and empty the hardware buffer later (in a thread in the buffer class) we can still overflow the HW buffer easily while the thread gets blocked elsewhere. It should be put in the FW buffer ASAP, as in the ISR. What is the issue with this?

Reply:
> you added this function to both mbed drivers and custom can library?

This as by accident, this is a WIP

Reply:
Draft PRs were a new concept to me, would change this to one if i knew how

Reply:
It won't overflow because you set the the thread to a very high priority (ie. Realtime) without adding any waits. Functionally, it become the exact same as reading in the interrupt, except its cleaner cause it avoids needing to remove the mutex

Reply:
The thread can only be "blocked elsewhere" if elsewhere has  critical section, in which interrupts would also be blocked

Reply:
Yeah that makes sense :+1: